### PR TITLE
Upgrades to Specs2 version 3

### PR DESCRIPTION
**Pull request by fedefernandez**
This PR:

- Upgrades to Specs2 version 3
- Fixes Lint errors on console
- Moves to travis for CI

Please @juanpedromoreno could you review? Thanks

@javipacheco @franciscodr thoughts?

### CODE REVIEW

File: project/Versions.scala [hunk 1]
```diff
@@ -12,7 +12,7 @@ object Versions {
   val playServicesV = "6.5.87"
   val playJsonV = "2.3.4"
   val picassoV = "2.5.0"
-  val specs2V = "2.4.17"
-  val mockitoV = "3.0-M2"
+  val specs2V = "3.3.1"
```
Comment (javipacheco) [inline project/Versions.scala]: I think that we should call to this variable `specsV`

Comment (juanpedromoreno) [inline project/Versions.scala]: Upgrade to the latest version: 3.6.1

File: src/test/scala/com/fortysevendeg/scala/android/modules/forecast/model/JsonModelSpec.scala [hunk 2]
```diff
@@ -16,17 +16,28 @@
...
-  with JsonModelTestSupportTestSupport {
-  
+  extends BaseTestSpecification {
 
   "load and map sample json" should {
 
-    "return an ApiModel class with the right fields" in {
+    "return an ApiModel class with the right fields" in new BaseTestScope {
       
       import JsonImplicits._
```
Comment (juanpedromoreno) [inline src/test/scala/com/fortysevendeg/scala/android/modules/forecast/model/JsonModelSpec.scala]: Maybe this implicits import should be at the beginning of the file.

File: src/test/scala/com/fortysevendeg/scala/android/modules/forecast/ForecastServicesSpec.scala [hunk 3]
```diff
@@ -2,29 +2,33 @@ package com.fortysevendeg.scala.android.modules.forecast
...
+
+  }
 
 }
 
-class ForecastServicesSpec
-  extends Specification
-  with BaseTestSupport {
-  
+trait ForecastServiceData {
```
Comment (juanpedromoreno) [inline src/test/scala/com/fortysevendeg/scala/android/modules/forecast/ForecastServicesSpec.scala]: ForecastService*s*Data?

File: src/test/scala/com/fortysevendeg/scala/android/modules/forecast/ForecastServicesSpec.scala [hunk 4]
```diff
@@ -2,29 +2,33 @@ package com.fortysevendeg.scala.android.modules.forecast
...
 
-trait ForecastServiceMock
-  extends ForecastServices
-  with ContextWrapperTestSupport {
+trait ForecastServiceSpecification
+  extends BaseTestSpecification
+  with ContextWrapperContextTestSupport {
 
-  override def loadJsonUrl(latitude: Double, longitude: Double)(implicit context: ContextWrapper): String = "http://fake_url/"
+  trait ForecastServiceScope
```
Comment (juanpedromoreno) [inline src/test/scala/com/fortysevendeg/scala/android/modules/forecast/ForecastServicesSpec.scala]: ForecastService*s*Scope?

**Comment by juanpedromoreno** [comment]
Minor comments added and LGTM ;)

File: project/Versions.scala (hunk 1 above)
Comment (fedefernandez) [inline project/Versions.scala]: @javipacheco but the library is `specs2` isn't it?

Comment (juanpedromoreno) [inline project/Versions.scala]: Yeah, I think specs2 is fine :)

**Comment by fedefernandez** [comment]
Comments addressed, thanks @juanpedromoreno

**Pull request merged**